certbot/compat.py

```diff
@@ -199,3 +199,19 @@ def get_default_folder(folder_type):
         return LINUX_DEFAULT_FOLDERS[folder_type]
     # Windows specific
     return WINDOWS_DEFAULT_FOLDERS[folder_type]
+
+
+def normalize_path(path):
+    # type: (str) -> str
+    """
+    Replace unsupported characters in path for current OS by underscores.
+    :param path: the path to normalize
+    :return: the normalized path
```

Comment:
Can we specify the return type here as well by using `rtype`?

Specifying these types in multiple places is redundant, but it makes sure the type is added to our API docs.